// @ionic-native/aes-256/index.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { IonicNativePlugin, cordova } from '@ionic-native/core';
var AES256Original = /** @class */ (function (_super) {
    __extends(AES256Original, _super);
    function AES256Original() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    AES256Original.prototype.encrypt = function (secureKey, secureIV, data) { return cordova(this, "encrypt", {}, arguments); };
    AES256Original.prototype.decrypt = function (secureKey, secureIV, data) { return cordova(this, "decrypt", {}, arguments); };
    AES256Original.prototype.generateSecureKey = function (password) { return cordova(this, "generateSecureKey", {}, arguments); };
    AES256Original.prototype.generateSecureIV = function (password) { return cordova(this, "generateSecureIV", {}, arguments); };
    AES256Original.pluginName = "AES256";
    AES256Original.plugin = "cordova-plugin-aes256-encryption";
    AES256Original.pluginRef = "cordova.plugins.AES256";
    AES256Original.repo = "https://github.com/Ideas2IT/cordova-aes256";
    AES256Original.platforms = ["Android", "iOS"];
    AES256Original.install = "ionic cordova plugin add cordova-plugin-aes256-encryption";
    return AES256Original;
}(IonicNativePlugin));
var AES256 = new AES256Original();
export { AES256 };
