// @ngx-formly/ionic/esm2015/select/select.type.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component } from '@angular/core';
import { FieldType } from '@ngx-formly/core';
export class FormlyFieldSelect extends FieldType {
    constructor() {
        super(...arguments);
        this.defaultOptions = {
            templateOptions: { options: [] },
        };
    }
}
FormlyFieldSelect.decorators = [
    { type: Component, args: [{
                selector: 'formly-field-ion-select',
                template: `
    <ion-select
      [formControl]="formControl"
      [ionFormlyAttributes]="field"
      [multiple]="to.multiple"
      [interface]="to.interface"
    >
      <ion-select-option *ngFor="let option of to.options | formlySelectOptions: field | async" [value]="option.value">
        {{ option.label }}
      </ion-select-option>
    </ion-select>
  `,
                styles: [':host { display: inherit; }']
            }] }
];
if (false) {
    /** @type {?} */
    FormlyFieldSelect.prototype.defaultOptions;
}
