// @ngx-formly/ionic/fesm2015/ngx-formly-ionic-select.js
import { Component, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { FieldType, FormlyModule } from '@ngx-formly/core';
import { IonicModule } from '@ionic/angular';
import { FormlySelectModule as FormlySelectModule$1 } from '@ngx-formly/core/select';
import { FormlyFormFieldModule } from '@ngx-formly/ionic/form-field';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class FormlyFieldSelect extends FieldType {
    constructor() {
        super(...arguments);
        this.defaultOptions = {
            templateOptions: { options: [] },
        };
    }
}
FormlyFieldSelect.decorators = [
    { type: Component, args: [{
                selector: 'formly-field-ion-select',
                template: `
    <ion-select
      [formControl]="formControl"
      [ionFormlyAttributes]="field"
      [multiple]="to.multiple"
      [interface]="to.interface"
    >
      <ion-select-option *ngFor="let option of to.options | formlySelectOptions: field | async" [value]="option.value">
        {{ option.label }}
      </ion-select-option>
    </ion-select>
  `,
                styles: [':host { display: inherit; }']
            }] }
];
if (false) {
    /** @type {?} */
    FormlyFieldSelect.prototype.defaultOptions;
}

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class FormlySelectModule {
}
FormlySelectModule.decorators = [
    { type: NgModule, args: [{
                declarations: [FormlyFieldSelect],
                imports: [
                    CommonModule,
                    ReactiveFormsModule,
                    IonicModule,
                    FormlyFormFieldModule,
                    FormlySelectModule$1,
                    FormlyModule.forChild({
                        types: [
                            {
                                name: 'select',
                                component: FormlyFieldSelect,
                                wrappers: ['form-field'],
                            },
                        ],
                    }),
                ],
            },] }
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { FormlyFieldSelect, FormlySelectModule };
