// @ngx-formly/ionic/fesm2015/ngx-formly-ionic-datetime.js
import { Component, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FieldType, FormlyModule } from '@ngx-formly/core';
import { ReactiveFormsModule } from '@angular/forms';
import { IonicModule } from '@ionic/angular';
import { FormlyFormFieldModule } from '@ngx-formly/ionic/form-field';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class FormlyFieldDatetime extends FieldType {
}
FormlyFieldDatetime.decorators = [
    { type: Component, args: [{
                selector: 'formly-field-ion-datetime',
                template: `
    <ion-datetime
      [displayFormat]="to.displayFormat"
      [pickerFormat]="to.pickerFormat"
      [min]="to.min"
      [max]="to.max"
      [formControl]="formControl"
      [ionFormlyAttributes]="field"
    >
    </ion-datetime>
  `
            }] }
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class FormlyDatetimeModule {
}
FormlyDatetimeModule.decorators = [
    { type: NgModule, args: [{
                declarations: [FormlyFieldDatetime],
                imports: [
                    CommonModule,
                    ReactiveFormsModule,
                    IonicModule,
                    FormlyFormFieldModule,
                    FormlyModule.forChild({
                        types: [
                            {
                                name: 'datetime',
                                component: FormlyFieldDatetime,
                                wrappers: ['form-field'],
                            },
                        ],
                    }),
                ],
            },] }
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { FormlyDatetimeModule, FormlyFieldDatetime as ɵa };
