// @ngx-formly/primeng/fesm5/ngx-formly-primeng-form-field.js
import { Component, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FieldWrapper, FormlyModule } from '@ngx-formly/core';
import { ReactiveFormsModule } from '@angular/forms';
import { __extends } from 'tslib';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var FormlyWrapperFormField = /** @class */ (function (_super) {
    __extends(FormlyWrapperFormField, _super);
    function FormlyWrapperFormField() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    FormlyWrapperFormField.decorators = [
        { type: Component, args: [{
                    selector: 'formly-wrapper-primeng-form-field',
                    template: "\n    <div *ngIf=\"to.label && to.hideLabel !== true\" class=\"ui-widget\">\n      <label [for]=\"id\">\n        {{ to.label }}\n        <span *ngIf=\"to.required && to.hideRequiredMarker !== true\">*</span>\n      </label>\n    </div>\n    <ng-container #fieldComponent></ng-container>\n\n    <div class=\"ui-message ui-messages-error\" *ngIf=\"showError\">\n      <formly-validation-message [field]=\"field\"></formly-validation-message>\n    </div>\n  ",
                    styles: ["\n      .ui-messages-error {\n        margin: 0;\n        margin-top: 4px;\n      }\n    "]
                }] }
    ];
    return FormlyWrapperFormField;
}(FieldWrapper));

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var FormlyFormFieldModule = /** @class */ (function () {
    function FormlyFormFieldModule() {
    }
    FormlyFormFieldModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [FormlyWrapperFormField],
                    imports: [
                        CommonModule,
                        ReactiveFormsModule,
                        FormlyModule.forChild({
                            wrappers: [
                                {
                                    name: 'form-field',
                                    component: FormlyWrapperFormField,
                                },
                            ],
                        }),
                    ],
                },] }
    ];
    return FormlyFormFieldModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { FormlyFormFieldModule, FormlyWrapperFormField as ɵa };
