// @ngx-formly/ionic/esm2015/datetime/datetime.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormlyModule } from '@ngx-formly/core';
import { ReactiveFormsModule } from '@angular/forms';
import { IonicModule } from '@ionic/angular';
import { FormlyFormFieldModule } from '@ngx-formly/ionic/form-field';
import { FormlyFieldDatetime } from './datetime.type';
export class FormlyDatetimeModule {
}
FormlyDatetimeModule.decorators = [
    { type: NgModule, args: [{
                declarations: [FormlyFieldDatetime],
                imports: [
                    CommonModule,
                    ReactiveFormsModule,
                    IonicModule,
                    FormlyFormFieldModule,
                    FormlyModule.forChild({
                        types: [
                            {
                                name: 'datetime',
                                component: FormlyFieldDatetime,
                                wrappers: ['form-field'],
                            },
                        ],
                    }),
                ],
            },] }
];
