// @ngx-formly/ionic/esm2015/form-field/form-field.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormlyModule } from '@ngx-formly/core';
import { ReactiveFormsModule } from '@angular/forms';
import { IonicModule } from '@ionic/angular';
import { FormlyWrapperFormField } from './form-field.wrapper';
import { IonFormlyAttributes } from './formly.attributes';
export class FormlyFormFieldModule {
}
FormlyFormFieldModule.decorators = [
    { type: NgModule, args: [{
                declarations: [FormlyWrapperFormField, IonFormlyAttributes],
                exports: [IonFormlyAttributes],
                imports: [
                    CommonModule,
                    ReactiveFormsModule,
                    IonicModule,
                    FormlyModule.forChild({
                        wrappers: [
                            {
                                name: 'form-field',
                                component: FormlyWrapperFormField,
                            },
                        ],
                    }),
                ],
            },] }
];
