// @ngx-formly/ionic/fesm5/ngx-formly-ionic-slider.js
import { Component, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FieldType, FormlyModule } from '@ngx-formly/core';
import { ReactiveFormsModule } from '@angular/forms';
import { IonicModule } from '@ionic/angular';
import { FormlyFormFieldModule } from '@ngx-formly/ionic/form-field';
import { __extends } from 'tslib';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var FormlyFieldSlider = /** @class */ (function (_super) {
    __extends(FormlyFieldSlider, _super);
    function FormlyFieldSlider() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    FormlyFieldSlider.decorators = [
        { type: Component, args: [{
                    selector: 'formly-field-ion-range',
                    template: "\n    <ion-range [min]=\"to.min\" [max]=\"to.max\" [formControl]=\"formControl\" [ionFormlyAttributes]=\"field\">\n      <ion-label slot=\"start\">{{ to.min }}</ion-label>\n      <ion-label slot=\"end\">{{ to.max }}</ion-label>\n    </ion-range>\n  ",
                    styles: [':host { display: inherit; }']
                }] }
    ];
    return FormlyFieldSlider;
}(FieldType));

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var FormlySliderModule = /** @class */ (function () {
    function FormlySliderModule() {
    }
    FormlySliderModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [FormlyFieldSlider],
                    imports: [
                        CommonModule,
                        ReactiveFormsModule,
                        IonicModule,
                        FormlyFormFieldModule,
                        FormlyModule.forChild({
                            types: [
                                { name: 'slider', component: FormlyFieldSlider, wrappers: ['form-field'] },
                                { name: 'range', extends: 'slider' },
                            ],
                        }),
                    ],
                },] }
    ];
    return FormlySliderModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { FormlySliderModule, FormlyFieldSlider as ɵa };
