// @ngx-formly/ionic/fesm5/ngx-formly-ionic-datetime.js
import { Component, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FieldType, FormlyModule } from '@ngx-formly/core';
import { ReactiveFormsModule } from '@angular/forms';
import { IonicModule } from '@ionic/angular';
import { FormlyFormFieldModule } from '@ngx-formly/ionic/form-field';
import { __extends } from 'tslib';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var FormlyFieldDatetime = /** @class */ (function (_super) {
    __extends(FormlyFieldDatetime, _super);
    function FormlyFieldDatetime() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    FormlyFieldDatetime.decorators = [
        { type: Component, args: [{
                    selector: 'formly-field-ion-datetime',
                    template: "\n    <ion-datetime\n      [displayFormat]=\"to.displayFormat\"\n      [pickerFormat]=\"to.pickerFormat\"\n      [min]=\"to.min\"\n      [max]=\"to.max\"\n      [formControl]=\"formControl\"\n      [ionFormlyAttributes]=\"field\"\n    >\n    </ion-datetime>\n  "
                }] }
    ];
    return FormlyFieldDatetime;
}(FieldType));

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var FormlyDatetimeModule = /** @class */ (function () {
    function FormlyDatetimeModule() {
    }
    FormlyDatetimeModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [FormlyFieldDatetime],
                    imports: [
                        CommonModule,
                        ReactiveFormsModule,
                        IonicModule,
                        FormlyFormFieldModule,
                        FormlyModule.forChild({
                            types: [
                                {
                                    name: 'datetime',
                                    component: FormlyFieldDatetime,
                                    wrappers: ['form-field'],
                                },
                            ],
                        }),
                    ],
                },] }
    ];
    return FormlyDatetimeModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { FormlyDatetimeModule, FormlyFieldDatetime as ɵa };
