// @ngx-formly/ionic/fesm5/ngx-formly-ionic-toggle.js
import { Component, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FieldType, FormlyModule } from '@ngx-formly/core';
import { ReactiveFormsModule } from '@angular/forms';
import { IonicModule } from '@ionic/angular';
import { FormlyFormFieldModule } from '@ngx-formly/ionic/form-field';
import { __extends } from 'tslib';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var FormlyFieldToggle = /** @class */ (function (_super) {
    __extends(FormlyFieldToggle, _super);
    function FormlyFieldToggle() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    FormlyFieldToggle.decorators = [
        { type: Component, args: [{
                    selector: 'formly-field-ion-toggle',
                    template: "\n    <ion-toggle [formControl]=\"formControl\" [ionFormlyAttributes]=\"field\"> </ion-toggle>\n  "
                }] }
    ];
    return FormlyFieldToggle;
}(FieldType));

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var FormlyToggleModule = /** @class */ (function () {
    function FormlyToggleModule() {
    }
    FormlyToggleModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [FormlyFieldToggle],
                    imports: [
                        CommonModule,
                        ReactiveFormsModule,
                        IonicModule,
                        FormlyFormFieldModule,
                        FormlyModule.forChild({
                            types: [
                                {
                                    name: 'toggle',
                                    component: FormlyFieldToggle,
                                    wrappers: ['form-field'],
                                },
                            ],
                        }),
                    ],
                },] }
    ];
    return FormlyToggleModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { FormlyToggleModule, FormlyFieldToggle as ɵa };
